app/src/main/java/edu/mills/findeatfood/MainActivity.java

```diff
@@ -1,13 +1,314 @@
 package edu.mills.findeatfood;
 
 import android.app.Activity;
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.app.FragmentTransaction;
+import android.content.res.Configuration;
 import android.os.Bundle;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.Toast;
 
-public class MainActivity extends Activity {
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * The top-level activity for the app FindEatFood. This activity is
+ * associated with the navigation drawer and its fragments, from which the user
+ * can search for recipes from an ingredient list. The app returns a list of
+ * recipes from an API if matches are found. The recipes can be added
+ * to favorites and stored in a database.
+ */
+
+public class MainActivity extends Activity
+        implements ResultsFragment.ResultsListListener, DietaryFragment.OnDietOptionsSelectedListener {
+
+    private String[] titles;
+    private ListView drawerList;
+    private DrawerLayout drawerLayout;
+    private ActionBarDrawerToggle drawerToggle;
+    private int currentPosition = 0;
+    private String[] ingredients;
+    /**
```

Comment:
Good comments. I know it's a pain. In normal environments, you won't have to write javadoc for package-private members.